### DISCUSSION TITLE
LazyVim Licensing

### DISCUSSION THREAD

**Discussion by mehalter** [LazyVim/LazyVim#1018] [General]
Hey, I just wanted to ask about the current licensing of LazyVim. It's currently licensed under Apache but has some downstream dependencies (plugins) that are GPL licensed. This will traverse through things that depend on them so I think that LazyVim is actually GPL licensed.

**Comment by folke** [comment]
LazyVim would still be Apache 2.0 in that case according to ChatGPT :)

> LazyVim is indeed licensed under the Apache 2.0 license. This is because the LazyVim project itself, as it's distributed, does not include any other code except its own, which is under Apache.

> The GPL plugins you mentioned are not bundled with our project. Instead, LazyVim configures a package manager to download and use these plugins. Moreover, LazyVim does not have a hard dependency on these plugins. All of these plugins can be disabled by users according to their preferences, and LazyVim will function without them.

> The GPL license mainly applies to distribution, and since we're not distributing the GPL-licensed plugins with our software, we believe that this doesn't cause LazyVim to be covered by the GPL.

**Reply by mehalter** [reply]
okie dokie. I guess I was considering having the code automatically downloaded (if the user doesn't manually disables them) as "included in the distribution"

**Reply by folke** [reply]
That's actually not even LazyVim. LazyVim itself is also just a plugin.

People just add LazyVim themselves as a plugin, or use the [starter](https://github.com/LazyVim/starter) which pre-configures the LazyVim plugin with lazy.nvim
